### ISSUE TITLE
Getting the CV's functional form

### ISSUE THREAD

**Issue by BSB47** [luigibonati/mlcolvar#169]
Hi there,

Thank you for publishing this code. It has been very interesting to play around with.

I'm trying to figure out whether it's possible to get the functional form of the MLCV out of the trained model, particularly the Deep-TICA (but if possible, others as well). Is this possible? So far I'm only able to get the CV values of different conformations for the top `n` CVs ranked by eigenvalue while using Deep-TICA loss.

I'm interested in this because I'd like to perform biased simulations e.g., umbrella sampling where it would be necessary to specify a priori some conformation at a given CV value (i.e., I'd like to reverse engineer a conformation from a CV value). So I'd like to have the CV specified as a function of the descriptors, which could then be fed to a MD program. If I understand correctly, the PLUMED module you wrote is calculating the CV value on-the-fly as the simulation is running, which is sufficient for OPES but not umbrella sampling.

Thank you!

**Comment by jintuzhang** [comment]
Hi,

I didn't get your requirements clearly, but from the task you described, you would like to evaluate the CV values for some existing trajectories, and then select conformations within some given CV ranges for the US simulations, right? If that were the task you would like to finish, you could use the [`plumed driver`](https://www.plumed.org/doc-v2.9/user-doc/html/driver.html) functionality.

**Comment by BSB47** [comment]
You have described the task partially. What I would like is to tell a MD program that is not plumed, such as OpenMM, to perform a biased simulation using the MLCV from your models. In order to do so, I need to express the MLCV as a function of the atom coordinates (you probably know this already). Is there a way to obtain such a function?

**Comment by luigibonati** [comment]
Hi,

The functional form of the MLCV, particularly Deep-TICA, is expressed as a composition of several linear and non-linear functions in the form of a neural network. While you could, in principle, extract the network weights and re-implement the model in another software, this approach is generally discouraged. 

If your goal is to use the MLCV in OpenMM, you have a few options:

1. **Use the PLUMED Interface**: OpenMM can be coupled with PLUMED, which allows on-the-fly evaluation of the MLCV during simulations. This is the recommended approach since PLUMED already supports MLCVs.
2. **Custom OpenMM Interface**: If you want to avoid using PLUMED, you would need to write a custom OpenMM interface that loads a JIT compiled Torchscript model, which you can use to evaluate the CV. Note that in this case you would need to calculate also the input features within OpenMM. 
3. **Distilling the MLCV into a Simpler Model:** If you need an explicit functional form, you might consider approximating the MLCV with a simpler, interpretable model. For example, you can use sparse regression techniques (such as LASSO) to fit a reduced model that captures the essential features of the learned MLCV. Check out the tutorial on sparse linear models here: https://mlcolvar.readthedocs.io/en/stable/notebooks/tutorials/expl_lasso.html.

Hope this helps

Luigi
